packages/@glimmer/runtime/lib/compiled/opcodes/content.ts

```diff
@@ -87,7 +96,11 @@ APPEND_OPCODES.add(Op.AppendHTML, (vm) => {
   let reference = check(vm.stack.pop(), CheckReference);
 
   let rawValue = valueForRef(reference);
-  let value = isEmpty(rawValue) ? '' : String(rawValue);
+  let value = isEmpty(rawValue)
+    ? ''
+    : isTrustedHTML(rawValue)
+      ? (rawValue as string)
```

Comment:
This cast seems suspect

Reply:
I would think, if we made it here, we should have already checked and unwrapped the value?

Reply:
We can modify whole chain of functions to accept `string | TrustedHTML` type, but I thought we should keep it when decide to fully migrate glimmer to Trusted Types.

Reply:
I've added  TrustedHTML for appendHTML methods, let's see if we have any issues

Reply:
Yeah, that's what I am trying to say, if you follow the types it'll lead you to the places where this doesn't just work. We can certainly consider changing/adding the requirement to simple-dom, we rarely do, but if this is a platform feature I think we can make a case for it; ultimately if you need it to make it all the way to the DOM (to satisfy CSP requirements, for example), then I think that's what this would entail. To just make it not error, we can unwrap it, but sounds like that may not be enough for you.

Reply:
https://github.com/ember-fastboot/simple-dom/blob/master/packages/%40simple-dom/interface/index.d.ts#L99
html should be unknown, not string. Since insertAdjacentHTML can accept any value and stringify it.

Reply:
simple-dom (as the README pointed out) isn't trying to model the entire DOM API, it's trying to model a small subset of the functionality that we actually need, as little as possible to make it work for our SSR use cases, it deliberately does not try to support all the features

Reply:
For example, it also doesn't have `.innerHTML = `, because just `insertAdjacentHTML` was enough to cover all the use cases we have (and in this case, it's a blessing because otherwise we would have to implement the TrustedHTML support in multiple places)

Reply:
that passed validation without modification of 
https://github.com/glimmerjs/glimmer-vm/blob/1cb493b128a9c6704482a1775a478aac20259935/packages/%40glimmer/runtime/lib/dom/operations.ts#L69
Do you want me to do other changes?

Reply:
Yes, we definitely still have to plumb this through to simple-dom. I am actually not sure why TypeScript is not catching the TypeError, but essentially the problem is you changed the interface without changing the implementations, so for example, in places like these, it's all going to be problematic:

https://github.com/glimmerjs/glimmer-vm/blob/main/packages/%40glimmer/runtime/lib/dom/operations.ts#L69-L74

https://github.com/glimmerjs/glimmer-vm/blob/1cb493b128a9c6704482a1775a478aac20259935/packages/%40glimmer/node/lib/node-dom-helper.ts#L23-L25

The code is a bit odd, because the primary implementation of `insertHTMLBefore` is on a class that doesn't claim to implement the interface you modified, so no type error there. But this extends and implement it, so it should be caught there, but it wasn't: https://github.com/glimmerjs/glimmer-vm/blob/main/packages/%40glimmer/runtime/lib/dom/api.ts#L18

Anyway missing type errors aside, there are some real, a bit tedious but solvable problems, there are also a few more things to think through around attributes (and then the grunt work to actually plumb it through, get simple-dom released, etc), so it may be a while before we can land this.

Reply:
I've changed it back to not change interface of insertHTMLBefore.

Reply:
If you think it will take a while, I suggest to taking small steps and in this PR just stringify TrusteHTML, but render it as HTML.

Reply:
> more things to think through around attributes
Let me explain why I'm against error on rendering TrustedHTML on attribute context.

I've tested LinkedIn where we use SafeString in attribute context. Almost everywhere, it's related to our i18n system. Every i18n string is like small HTML template where all params HTML encoded, not every i18n string contains HTML tags, but potentially every string can contain HTML entities, like `&nbsp;`. For this reason i18n helper uses htmlSafe for every i18n string. And obviously same htmlSafe strings used in attribute context. If wea are going to replace SafeString with TrustedHTML it will not change anything. Throwing errors would completely prevent migration to TrustedHTML.

I would recommend to keep same behavior as SafeString, and in best case don't stringify TrusteHTML object.

Another point is DOM API, some attributes require TrustedHTML to work, for example iframe.srcdoc
```
trustedTypes.getAttributeType('iframe', 'srcdoc') // TrustedHTML
```
https://developer.mozilla.org/en-US/docs/Web/API/TrustedTypePolicyFactory/getAttributeType

Reply:
Thanks for walking through those use cases. That's helpful, and I'm fine with not throwing